demos/python_demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.py

```diff
@@ -15,241 +15,581 @@
+
+    def __len__(self):
+        return len(self.palette)
+
+
+class Model:
+    def __init__(self, ie, xml_file_path, bin_file_path=None,
+                 device='CPU', plugin_config={}, max_num_requests=1,
+                 results={}, caught_exceptions=[]):
+        self.ie = ie
+        log.info('Reading network from IR...')
+        if bin_file_path is None:
+            bin_file_path = osp.splitext(xml_file_path)[0] + '.bin'
+        self.net = ie.read_network(model=xml_file_path, weights=bin_file_path)
+
+        log.info('Loading network to plugin...')
+        if 'CPU' in device:
+            self.check_cpu_support(ie, self.net)
+        self.max_num_requests = max_num_requests
+        self.exec_net = ie.load_network(network=self.net, device_name=device, config=plugin_config, num_requests=max_num_requests)
+
+        self.requests = self.exec_net.requests
+        self.empty_requests = deque(self.requests)
+        self.completed_request_results = results
+        self.callback_exceptions = caught_exceptions
+        self.event = threading.Event()
+
+    @staticmethod
+    def check_cpu_support(ie, net):
+        log.info('Check that all layers are supported...')
+        supported_layers = ie.query_network(net, 'CPU')
+        not_supported_layers = [l for l in net.layers.keys() if l not in supported_layers]
+        if len(not_supported_layers) != 0:
+            unsupported_info = '\n\t'.join('{} ({} with params {})'.format(layer_id,
+                                                                           net.layers[layer_id].type,
+                                                                           str(net.layers[layer_id].params))
+                                           for layer_id in not_supported_layers)
+            log.warning('Following layers are not supported '
+                        'by the CPU plugin:\n\t{}'.format(unsupported_info))
+            log.warning('Please try to specify cpu extensions library path.')
+            raise ValueError('Some of the layers are not supported.')
+
+    def unify_inputs(self, inputs):
+        if not isinstance(inputs, dict):
+            if not isinstance(inputs, (list, tuple)):
+                inputs = [inputs]
+            if len(inputs) == 1:
+                inputs_dict = {next(iter(self.net.input_info)): inputs[0]}
+            else:
+                raise ValueError
+        else:
+            inputs_dict = inputs
+        return inputs_dict
+
+    def preprocess(self, inputs):
+        meta = {}
+        return inputs, meta
+
+    def postprocess(self, outputs, meta):
+        return outputs
+
+    def request_callback(self, status, callback_args):
+        request, frame_id, frame_meta = callback_args
+        try:
+            if status != 0:
+                raise RuntimeError('Infer Request has returned status code {}'.format(status))
+            raw_outputs = {key: blob.buffer for key, blob in request.output_blobs.items()}
+            outputs = None
```

Comment:
That var has become pointless with moving of callback.

Reply:
Removed.